### PR TITLE
Application document sort order

### PR DESCRIPTION
**Pull request by CEBergin**
Change to search to allow multiple sort orders to be passed in a comma delimited method in order to handle ticket: https://bcmines.atlassian.net/browse/EE-808
(The sort order for the Ticket was required to be Sort Order, Date, Alphanumeric )
Previously multiple sorts were not being handled.

Due to the possibility of being presented with multiple sortBy's making the sortBy change it's type there are two ways of handling these properties.

### CODE REVIEW

File: api/aggregators/searchAggregator.js [hunk 1]
```diff
@@ -108,10 +108,26 @@ exports.createMatchAggr = async (schemaName, projectId, keywords, caseSensitive,
  */
 exports.createSortingPagingAggr = function(schemaName, sortValues, sortField, sortDirection, pageNum, pageSize) {
   const searchResultAggregation = [];
-  let defaultTwoSorts = false;
+  let dateHandling = false;
+  console.log(Object.keys(sortValues).length);
```
Comment (dhlevi) [inline api/aggregators/searchAggregator.js]: Remove the console.log. If we need the logging, replace with a default logger

File: api/aggregators/searchAggregator.js [hunk 2]
```diff
@@ -108,10 +108,26 @@ exports.createMatchAggr = async (schemaName, projectId, keywords, caseSensitive,
...
+  if (sortField && sortField.includes(",") || Object.keys(sortValues).length > 1){
+    //sort will have multiple values passed
+    if (sortField.includes("datePosted") || sortValues.hasOwnProperty("datePosted")){
```
Comment (dhlevi) [inline api/aggregators/searchAggregator.js]: Is this condition only true for datePosted, or could there be issues with all dates>?

File: api/aggregators/searchAggregator.js [hunk 3]
```diff
@@ -121,7 +137,7 @@ exports.createSortingPagingAggr = function(schemaName, sortValues, sortField, so
 
   // We don't want to have sort in the aggregation if the front end doesn't need sort.
   if (sortField && sortDirection) {
-    if (defaultTwoSorts){
+     if(dateHandling){
```
Comment (dhlevi) [inline api/aggregators/searchAggregator.js]: Sort of implies this is a handler for any sorted date field, even though it's specific to sorting with datePosted. MIght want to rename for clarity

File: api/aggregators/searchAggregator.js (hunk 2 above)
Comment (CEBergin) [reply api/aggregators/searchAggregator.js]: datePosted is the only column we sort on currently and the handling is essentially just for us. We could in the future expand this to handle all types of dates if it becomes necessary.

File: api/aggregators/searchAggregator.js [hunk 4]
```diff
@@ -108,10 +108,25 @@ exports.createMatchAggr = async (schemaName, projectId, keywords, caseSensitive,
...
 exports.createSortingPagingAggr = function(schemaName, sortValues, sortField, sortDirection, pageNum, pageSize) {
   const searchResultAggregation = [];
-  let defaultTwoSorts = false;
+  let datePostedHandlingTruncating = false;
+  if (sortField && sortField.includes(",") || Object.keys(sortValues).length > 1){
+    //sort will have multiple values passed
+    if (sortField.includes("datePosted") || sortValues.hasOwnProperty("datePosted")){
+      //datePosted is too specfic(in it's time) and needs the truncated form of date, can be expanded if other dates are required to be truncated
+      tempSortValues = {};
+      for (let property in sortValues){
```
Comment (KitArmstrong) [inline api/aggregators/searchAggregator.js]: Can `sortValues` be blank if `sortField` is an array? Or if `sortField` exists then should there also be `sortValues`?

File: api/aggregators/searchAggregator.js [hunk 5]
```diff
@@ -108,10 +108,25 @@ exports.createMatchAggr = async (schemaName, projectId, keywords, caseSensitive,
  */
 exports.createSortingPagingAggr = function(schemaName, sortValues, sortField, sortDirection, pageNum, pageSize) {
   const searchResultAggregation = [];
-  let defaultTwoSorts = false;
+  let datePostedHandlingTruncating = false;
+  if (sortField && sortField.includes(",") || Object.keys(sortValues).length > 1){
+    //sort will have multiple values passed
+    if (sortField.includes("datePosted") || sortValues.hasOwnProperty("datePosted")){
```
Comment (KitArmstrong) [inline api/aggregators/searchAggregator.js]: Calling `hasOwnProperty` in this way will fail linting. You have it correct further down.

File: api/aggregators/searchAggregator.js (hunk 4 above)
Comment (CEBergin) [reply api/aggregators/searchAggregator.js]: No, Yes.

File: api/aggregators/searchAggregator.js (hunk 5 above)
Comment (CEBergin) [reply api/aggregators/searchAggregator.js]: Changed.

**Pull request merged**